File: lab2/source/more.cpp
#include "operations.h"
#include "factory.h"

class More : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        Empty_check(stack);
        int previous_top = stack.top();
        stack.pop();
        Empty_check(stack);
        int res = 0;
        if(stack.top() > previous_top) res = 1;
        stack.pop();
        stack.push(res);
    }

};

namespace {
    Operations* createMore() {
        return new More;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string(">"), createMore);
}

File: jaba/lab1/src/main/java/org/example/More.java
package org.example;

import java.io.OutputStream;
import java.util.Scanner;
import java.util.Stack;

public class More extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s) {
        emptyCheck(stack);
        int previous_top = stack.pop();
        emptyCheck(stack);
        int res = 0;
        if(stack.pop() > previous_top) res = 1;
        stack.push(res);
    }
}
